src/Components/3DV/SceneView.tsx

```diff
@@ -1653,6 +1660,205 @@ function SceneView(props: ISceneViewProps, ref) {
+
+                            attachedMesh.position.x = transform.position.x;
+                            attachedMesh.position.y = transform.position.y;
+                            attachedMesh.position.z = transform.position.z;
+                        } else {
+                            gizmoTransformItemRef.current = {
+                                meshIds: deepCopy(meshIds),
+                                parentMeshId: parentMeshId,
+                                original: originalTransform,
+                                transform: originalTransform
+                            };
+                        }
+
+                        setGizmoTransformItem(
+                            gizmoTransformItemRef.current.transform
+                        );
+                    });
+
+                    // update the gizmoTransformItem (allows builder panel to display new position/rotation)
+                    const updateTransform = () => {
+                        const attachedMesh =
+                            gizmoManager.gizmos.positionGizmo.attachedMesh;
+
+                        // this is the main place where transforms get set, so round here
+                        gizmoTransformItemRef.current.transform = {
+                            position: {
+                                x: Math.round(attachedMesh.position.x),
+                                y: Math.round(attachedMesh.position.y),
+                                z: Math.round(attachedMesh.position.z)
+                            },
+                            rotation: {
+                                x: Number(attachedMesh.rotation.x.toFixed(2)),
+                                y: Number(attachedMesh.rotation.y.toFixed(2)),
+                                z: Number(attachedMesh.rotation.z.toFixed(2))
+                            }
+                        };
+                        setGizmoTransformItem(
+                            gizmoTransformItemRef.current.transform
+                        );
+                    };
+
+                    // on drag end for both position and rotation gizmos, update transform
+                    // updating on every frame is too much for react to handle
+                    const positionGizmo = gizmoManager.gizmos.positionGizmo;
+                    positionGizmo.onDragEndObservable.add(() => {
+                        updateTransform();
+                    });
+
+                    const rotationGizmo = gizmoManager.gizmos.rotationGizmo;
+                    rotationGizmo.onDragEndObservable.add(() => {
+                        updateTransform();
+                    });
+                }
+            } catch {
+                console.warn('unable to add gizmo to mesh');
+            }
+            if (debugLogging) {
+                console.timeEnd('adding gizmo to meshes');
+            }
+        }
+
+        return () => {
+            debugLog('debug', 'Mesh gizmo cleanup');
```

Comment:
A common problem with SceneView is cacheing objects when SceneView is recycled. I suspect if you switch models (scenes) you will get a crash as the Babylon context will be reset, but the code will try and use the gizmo stuff from the previous model. See the ADTViewer/ZoomAndColor story where you can switch scenes to test this (add some gizmo stuff to this story to test).

I think you need to add some cleanup code to this effect (eg clear gizmoManagerRef and gizmoTransformItemRef so that they get recreated)

Reply:
I checked it out, and clearing gizmoManagerRef and gizmoTransformItemRef actually causes issues with switching in and out of the tab that enables the transform. Based on our earlier conversation and on the ADTViewer story, though, I know that the user's application doesn't always consume the builder -- what if I added the scene as a dependency to the useEffect?

Reply:
done